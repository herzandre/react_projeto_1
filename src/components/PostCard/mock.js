export const postCardPropsMock = {
    title: 'title-test',
    body: 'body-test',
    id: 1,
    cover: 'img/img.png'
}