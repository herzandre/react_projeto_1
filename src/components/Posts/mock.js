export const postsPropsMock =  {
    posts: [
        {
            id: 1,
            title: 'title-test 1',
            body: 'body-test 1',
            cover: 'img/img1.jpg'
        },
        {
            id: 2,
            title: 'title-test 2',
            body: 'body-test 2',
            cover: 'img/img2.jpg'
        },
        {
            id: 3,
            title: 'title-test 3',
            body: 'body-test 3',
            cover: 'img/img3.jpg'
        }
    ]
}