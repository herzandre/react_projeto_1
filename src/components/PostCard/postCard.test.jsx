import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly"', () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText('title-test')).toBeInTheDocument();

    expect(screen.getByRole('img', { name: 'title-test' })).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: 'title-test 1' })).toBeInTheDocument();

    expect(screen.getByText('body-test')).toBeInTheDocument();
  });

  it('should match spanshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
