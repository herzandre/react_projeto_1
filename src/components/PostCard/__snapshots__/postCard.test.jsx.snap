// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match spanshot 1`] = `
<div
  class="post"
  id="1"
>
  <img
    alt="title-test"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
       
      title-test
       
      1
       
    </h2>
    <p>
       
      body-test
       
    </p>
  </div>
</div>
`;
